plugins/auth-backend/src/lib/resolvers/CatalogAuthResolverContext.ts

```diff
@@ -123,8 +124,14 @@ export class CatalogAuthResolverContext implements AuthResolverContext {
       const res = await this.catalogApi.getEntities({ filter }, { token });
       result = res.items;
     } else if ('filter' in query) {
+      const filter = lodash.cloneDeep(query.filter);
```

Comment:
Tbh I think it's a bit cleaner to achieve this using object spread instead, and mapping over the filter array to produce the new filters, while always passing a filter array to `getEntities`